### PR TITLE
Added support for SchemaChangeAdaptor to adapt upgrade steps in-flight.

### PR DESCRIPTION
**Pull request by jsimlo**
+ Renamed UpgradeConfiguration to UpgradeConfigAndContext.

### CODE REVIEW

**Comment by jsimlo** [comment]
[70.4% Coverage](https://sonarcloud.io/component_measures?id=org.alfasoftware%3Amorf-parent&pullRequest=328&metric=new_coverage&view=list)  Missing coverage is on trivial code.

File: morf-core/src/main/java/org/alfasoftware/morf/upgrade/SchemaChangeAdaptor.java [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+      return second.adapt(first.adapt(executeStatement));
+    }
+
+    @Override
+    public RenameTable adapt(RenameTable renameTable) {
+      return second.adapt(first.adapt(renameTable));
+    }
+
+    @Override
+    public ChangePrimaryKeyColumns adapt(ChangePrimaryKeyColumns renameTable) {
```
Comment (euano) [inline morf-core/src/main/java/org/alfasoftware/morf/upgrade/SchemaChangeAdaptor.java:248]: parameter name doesn't match type

File: morf-core/src/main/java/org/alfasoftware/morf/upgrade/SchemaChangeAdaptor.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+   * By no-op, we mean non-changing: the input is passed through as output.
+   */
+  public static final class NoOp implements SchemaChangeAdaptor {
+  }
+
+
+  /**
+   * Combines two {@link SchemaChangeAdaptor}s to run the second one on the results of the first one.
+   */
+  public static final class Combining implements SchemaChangeAdaptor {
```
Comment (euano) [inline morf-core/src/main/java/org/alfasoftware/morf/upgrade/SchemaChangeAdaptor.java:182]: nice

File: morf-core/src/main/java/org/alfasoftware/morf/upgrade/UpgradeConfigAndContext.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+   * Set of full names of upgrade step classes which should be executed non-parallel.
+   * No other other upgrade step will be running while listed upgrade steps are being executed.
+   * This impacts Graph-Based Upgrade only.
+   */
+  private Set<String> exclusiveExecutionSteps = Set.of();
+
+  /**
+   * A schema change adaptor to be used during upgrade.
+   */
+  private SchemaChangeAdaptor upgradeChangeAdaptor = new SchemaChangeAdaptor.NoOp();
```
Comment (euano) [inline morf-core/src/main/java/org/alfasoftware/morf/upgrade/UpgradeConfigAndContext.java:24]: pretty minor, but the variable being named upgradeChangeAdaptor when the type is schemaChangeAdaptor could perhaps add a slightly mild hint of momentary confusion that could be avoided

Comment (euano) [reply morf-core/src/main/java/org/alfasoftware/morf/upgrade/UpgradeConfigAndContext.java:24]: (same in public method on this class)

Comment (jsimlo) [reply morf-core/src/main/java/org/alfasoftware/morf/upgrade/UpgradeConfigAndContext.java:24]: Well spotted.

**Review by euano — APPROVED** [review]

**Pull request merged**